File: backend/src/MyDatingProfileHelper.Backend.Api/DTOs/GeneratedProfileHistoryDto.cs
csharp
namespace MyDatingProfileHelper.Backend.Api.DTOs
{
    /// <summary>
    /// DTO для представления истории сгенерированной анкеты в API.
    /// </summary>
    public class GeneratedProfileHistoryDto
    {
        /// <summary>
        /// Идентификатор записи истории.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Сгенерированный текст анкеты.
        /// </summary>
        public string GeneratedText { get; set; }

        /// <summary>
        /// Время генерации.
        /// </summary>
        public DateTime Timestamp { get; set; }
    }
}

File: app/src/main/java/com/jounik_projects/mydatingprofilehelper/data/network/model/GeneratedProfileHistoryDto.kt
package com.jounik_projects.mydatingprofilehelper.data.network.model

/**
 * Модель данных для истории сгенерированной анкеты, получаемая с бэкенда.
 */

data class GeneratedProfileHistoryDto(
    val id: Int, // Идентификатор записи истории
 val userProfileId: Int, // Идентификатор профиля пользователя
    val generatedText: String, // Текст сгенерированной анкеты
    val timestamp: String // Время генерации (в формате строки, как приходит с бэкенда)
    // Дополнительные поля, если есть в DTO бэкенда
)